test/verify/gemm_softmax_gemm_relu.cpp

```diff
@@ -41,17 +43,34 @@ struct gemm_softmax_gemm_relu : verify_program<gemm_softmax_gemm_relu>
         auto b1          = mm->add_parameter("3", m1_shape);
         std::vector<float> eights(m2_elements, 0.125);
         auto eight = mm->add_literal(migraphx::literal{m2_shape, eights});
-        std::vector<float> zeros(m2_elements, 0);
-        auto zero = mm->add_literal(migraphx::literal{m2_shape, zeros});
 
         b = mm->add_instruction(migraphx::make_op("transpose", {{"permutation", {0, 1, 3, 2}}}), b);
-        auto gemm1   = mm->add_instruction(migraphx::make_op("dot"), a, b);
-        auto scale   = mm->add_instruction(migraphx::make_op("mul"), gemm1, eight);
-        auto bias    = mm->add_instruction(migraphx::make_op("add"), scale, zero);
-        auto softmax = mm->add_instruction(migraphx::make_op("softmax", {{"axis", 3}}), bias);
+        auto gemm1 = mm->add_instruction(migraphx::make_op("dot"), a, b);
+        auto scale = mm->add_instruction(migraphx::make_op("mul"), gemm1, eight);
+
+        std::optional<migraphx::instruction_ref> add_bias{std::nullopt};
+        if constexpr(WithBias)
+        {
+            auto bias_strides = m1_shape.strides();
+            if(not WithStandardBiasShape)
+            {
+                const auto last_index{bias_strides.size() - 1};
+                std::swap(bias_strides[last_index], bias_strides[last_index - 1]);
```

Comment:
both `m1_shape` and `m2_shape` are same.  you can remove one of that. 

Also just FYI instead of manually calculating strides, migraphx has a utility called `shape::from_permutation`.

So for this case you can use it like 
`auto bias_shape = shape::from_permutation(m1_shape.type(), m1_shape.lens(), {0, 1, 3, 2})`

Reply:
Thanks for your suggestion. Done!